solidity/test/vault/TBTCVault.test.ts

```diff
@@ -439,4 +439,129 @@ describe("TBTCVault", () => {
       })
     })
   })
+
+  describe("onBalanceIncreased", () => {
+    const depositor1 = "0x30c371E0651B2Ff6062158ca1D95b07C7531c719"
+    const depositor2 = "0xb3464806d680722dBc678996F1670D19A42eA3e9"
+    const depositor3 = "0x6B9925e04bc46569d1F7362eD7f11539234f0aEc"
+
+    const depositedAmount1 = to1e18(19)
+    const depositedAmount2 = to1e18(11)
+    const depositedAmount3 = to1e18(301)
+
+    const totalDepositedAmount = to1e18(331) // 19 + 11 + 301
+
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    context("when called not by the Bank", () => {
+      it("should revert", async () => {
+        await expect(
+          vault
+            .connect(bridge)
+            .onBalanceIncreased([depositor1], [depositedAmount1])
+        ).to.be.revertedWith("Caller is not the Bank")
+      })
+    })
+
+    context("when called with no depositors", () => {
+      it("should revert", async () => {
+        await expect(
+          bank.connect(bridge).increaseBalanceAndCall(vault.address, [], [])
+        ).to.be.revertedWith("No depositors specified")
+      })
+    })
+
+    context(
+      "when depositors array has different length than amounts array",
```

Comment:
Would be good to have the opposite case (amounts.length > depositors.length) covered.

Reply:
https://github.com/keep-network/tbtc-v2/pull/94/commits/a4370e66f09b8da92eb7f14eb941b4c329a6baba